client/memory.cpp

```diff
@@ -30,6 +30,14 @@ using namespace RemoteCL;
 using namespace RemoteCL::Client;
 
 
+struct MappedPtr {
+	void* ptr;
+	size_t offset;
+	size_t size;
+};
+
+static std::map<cl_mem, MappedPtr> MappedObjects;
```

Comment:
This isn't thread-safe.
Tracking of mapping should be done as part of the Connection object, and registering mapping ought to be done with a LockedConnection to ensure thread safety.